components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/manager/FxaAccountManager.kt

```diff
@@ -865,10 +881,30 @@ open class FxaAccountManager(
                         // If we fail with a 401, then we know we have a legitimate authentication problem.
                         logger.info("Hit auth problem. Trying to recover.")
 
+                        // Ensure we clear any auth-relevant internal state, such as access tokens.
+                        account.authErrorDetected()
+
                         // Clear our access token cache; it'll be re-populated as part of the
                         // regular state machine flow if we manage to recover.
                         SyncAuthInfoCache(context).clear()
 
+                        // Circuit-breaker logic to protect ourselves from getting into endless authorization
+                        // check loops. If we determine that application is trying to check auth status too
+                        // frequently, drive the state machine into an unauthorized state.
```

Comment:
Driving to "unauthorized" seems reasonable here; it won't necessarily help the user to access their account data, but at least the rest of the browser will be useable.